api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.patientqueueing.api;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.openmrs.BaseOpenmrsObjectTest;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.patientqueueing.PatientQueueingConfig;
+import org.openmrs.module.patientqueueing.api.dao.PatientQueueingDao;
+import org.openmrs.module.patientqueueing.api.impl.PatientQueueingServiceImpl;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+
+import javax.validation.constraints.AssertTrue;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a unit test, which verifies logic in UgandaEMRSyncService. It doesn't extend
+ * BaseModuleContextSensitiveTest, thus it is run without the in-memory DB and Spring context.
+ */
+public class PatientQueueingServiceTest extends BaseModuleContextSensitiveTest {
+
+    protected static final String QUEUE_STANDARD_DATASET_XML = "org/openmrs/module/patientqueueing/standardTestDataset.xml";
+
+
+    @Before
+    public void initialize() throws Exception {
+        executeDataSet(QUEUE_STANDARD_DATASET_XML);
+    }
+
+    @InjectMocks
+    PatientQueueingServiceImpl patientQueueingService;
+
+    @Mock
+    PatientQueueingDao dao;
+
+    @Before
+    public void setupMocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void savePatientQueue_shouldReturnPatientQueueWithCompletedStatus() throws IOException {
```

Comment:
Are you testing savePatientQueue or completePatientQueue?

Reply:
This is testing completing Patient Queue. Will rename the method